### ISSUE TITLE
enableMarkdownInDescription does not work in parent elements

### ISSUE THREAD

**Issue by quazgar** [rjsf-team/react-jsonschema-form#3975]
### Prerequisites

- [X] I have searched the existing issues
- [X] I understand that providing a [SSCCE](http://sscce.org/) example is tremendously useful to the maintainers.
- [X] I have read the [documentation](https://rjsf-team.github.io/react-jsonschema-form/docs)
- [X] Ideally, I'm providing a [sample JSFiddle](https://jsfiddle.net), [Codesandbox.io](https://codesandbox.io) or preferably a [shared playground link](https://rjsf-team.github.io/react-jsonschema-form/) demonstrating the issue.

### What theme are you using?

material-ui

### Version

5.x / current playground

### Current Behavior

As demonstrated in [this playground instance](https://rjsf-team.github.io/react-jsonschema-form/#eyJmb3JtRGF0YSI6e30sInNjaGVtYSI6eyJ0aXRsZSI6IkEgbGlzdCBvZiB0YXNrcyIsInR5cGUiOiJvYmplY3QiLCJwcm9wZXJ0aWVzIjp7InRhc2tzIjp7ImRlc2NyaXB0aW9uIjoiKnZlcnkqIHNpbXBsZSIsInR5cGUiOiJvYmplY3QiLCJ0aXRsZSI6IlRhc2tzIiwicHJvcGVydGllcyI6eyJkZXRhaWxzIjp7InR5cGUiOiJzdHJpbmciLCJ0aXRsZSI6IlRhc2sgZGV0YWlscyIsImRlc2NyaXB0aW9uIjoiRW50ZXIgdGhlICp0YXNrKiBkZXRhaWxzIn19fX19LCJ1aVNjaGVtYSI6eyJ0YXNrcyI6eyJ1aTplbmFibGVNYXJrZG93bkluRGVzY3JpcHRpb24iOnRydWUsInVpOmRlc2NyaXB0aW9uIjoiTmV3ICpkZXNjcmlwdGlvbioiLCJkZXRhaWxzIjp7InVpOmVuYWJsZU1hcmtkb3duSW5EZXNjcmlwdGlvbiI6dHJ1ZSwidWk6d2lkZ2V0IjoidGV4dGFyZWEifX19LCJ0aGVtZSI6Im1hdGVyaWFsLXVpLTUiLCJsaXZlU2V0dGluZ3MiOnsic2hvd0Vycm9yTGlzdCI6InRvcCIsImV4cGVyaW1lbnRhbF9kZWZhdWx0Rm9ybVN0YXRlQmVoYXZpb3IiOnsiYXJyYXlNaW5JdGVtcyI6eyJwb3B1bGF0ZSI6InBvcHVsYXRlIiwibWVyZ2VFeHRyYURlZmF1bHRzIjpmYWxzZX0sImVtcHR5T2JqZWN0RmllbGRzIjoicG9wdWxhdGVBbGxEZWZhdWx0cyJ9fX0=), `enableMarkdownInDescription` only seems to work in leaf nodes, not in higher levels of the json schema tree.  

### Expected Behavior

`enableMarkdownInDescription` should work everywhere.

### Steps To Reproduce

Option 1: Look ath the playground example linked above.

Option 2:

1. Add `"ui:enableMarkdownInDescription": true` to the ui schema definition in an element which has children.
2. Add a description which should be rendered as Markdown.

### Environment

```markdown
Current playground.
```

**Comment by quazgar** [comment]
@dejanbelusic @heath-freenome pinging, because you seem to have been involved in #3665

**Comment by heath-freenome** [comment]
@quazgar Good catch. I'm really busy and I do not have time to make a fix. If @dejanbelusic is unable to help, would you be willing to provide a PR to fix this?

**Comment by quazgar** [comment]
I am pretty busy myself, but maybe I can find a customer who needs this and is willing to pay, or a colleague might find the time.

**Comment by dejanbelusic** [comment]
Hi @heath-freenome, I created a PR fixing this issue: https://github.com/rjsf-team/react-jsonschema-form/pull/4405, you can take a look!

Hi @quazgar, sorry it took me this long to tackle this issue. I wish I could have done it sooner, hope it still brings you value.

**Comment by dejanbelusic** [comment]
> [New playground instance](https://rjsf-team.github.io/react-jsonschema-form/#eyJmb3JtRGF0YSI6e30sInNjaGVtYSI6eyJ0aXRsZSI6IkEgbGlzdCBvZiB0YXNrcyIsInR5cGUiOiJvYmplY3QiLCJwcm9wZXJ0aWVzIjp7InRhc2tzIjp7ImRlc2NyaXB0aW9uIjoiKnZlcnkqIHNpbXBsZSwgYnV0IG5vIE1hcmtkb3duIiwidHlwZSI6Im9iamVjdCIsInRpdGxlIjoiVGFza3MiLCJwcm9wZXJ0aWVzIjp7ImRldGFpbHMiOnsidHlwZSI6InN0cmluZyIsInRpdGxlIjoiVGFzayBkZXRhaWxzIiwiZGVzY3JpcHRpb24iOiJEZXNjcmlwdGlvbiB3aGljaCByZW5kZXJzIE1hcmtkb3duICpjb3JyZWN0bHkqLiJ9LCJoYXNfTWFya2Rvd24iOnsidHlwZSI6ImJvb2xlYW4iLCJkZXNjcmlwdGlvbiI6IkFsc28gKm5vIE1hcmtkb3duKiJ9fX19fSwidWlTY2hlbWEiOnsidGFza3MiOnsidWk6ZW5hYmxlTWFya2Rvd25JbkRlc2NyaXB0aW9uIjp0cnVlLCJ1aTpkZXNjcmlwdGlvbiI6Ik5ldyAqZGVzY3JpcHRpb24qLCBzdGlsbCBubyBNYXJrZG93bi4iLCJkZXRhaWxzIjp7InVpOmVuYWJsZU1hcmtkb3duSW5EZXNjcmlwdGlvbiI6dHJ1ZSwidWk6d2lkZ2V0IjoidGV4dGFyZWEifSwiaGFzX21hcmtkb3duIjp7InVpOmVuYWJsZU1hcmtkb3duSW5EZXNjcmlwdGlvbiI6dHJ1ZX19fSwidGhlbWUiOiJtYXRlcmlhbC11aS01IiwibGl2ZVNldHRpbmdzIjp7InNob3dFcnJvckxpc3QiOiJ0b3AiLCJleHBlcmltZW50YWxfZGVmYXVsdEZvcm1TdGF0ZUJlaGF2aW9yIjp7ImFycmF5TWluSXRlbXMiOnsicG9wdWxhdGUiOiJwb3B1bGF0ZSIsIm1lcmdlRXh0cmFEZWZhdWx0cyI6ZmFsc2V9LCJhbGxPZiI6InNraXBEZWZhdWx0cyIsImVtcHR5T2JqZWN0RmllbGRzIjoicG9wdWxhdGVBbGxEZWZhdWx0cyJ9fX0=) which shows that markdown also does not seem to work for descriptions which render _above_ the input element (a boolean field in this case).
> 
> #### Schema
> ```json
> {
>   "title": "A list of tasks",
>   "type": "object",
>   "properties": {
>     "tasks": {
>       "description": "*very* simple, but no Markdown",
>       "type": "object",
>       "title": "Tasks",
>       "properties": {
>         "details": {
>           "type": "string",
>           "title": "Task details",
>           "description": "Description which renders Markdown *correctly*."
>         },
>         "has_Markdown": {
>           "type": "boolean",
>           "description": "Also *no Markdown*"
> }}}}}
> ```
> 
> #### UI schema
> ```json
> {
>   "tasks": {
>     "ui:enableMarkdownInDescription": true,
>     "ui:description": "New *description*, still no Markdown.",
>     "details": {
>       "ui:enableMarkdownInDescription": true,
>       "ui:widget": "textarea"
>     },
>     "has_markdown": {
>       "ui:enableMarkdownInDescription": true
> }}}
> ```

In the example you shared here, please note that in Shema you are using `has_Markdown`,  but in UI schema it says `has_markdown`. These values are case-sensitive and should be the same.  Please note that when testing the feature after my PR gets merged.

**Comment by quazgar** [comment]
Thank you, this seems to be fixed, at least in the playground instance.
